contracts/SwapAndBridge.sol

```diff
@@ -291,4 +300,78 @@ contract UniversalSwapAndBridge is SwapAndBridgeBase {
             acrossInputToken
         );
     }
+
+    /**
+     * @notice Swaps an EIP-2612 token on this chain via specified router before submitting Across deposit atomically.
+     * Caller can specify their slippage tolerance for the swap and Across deposit params.
+     * @dev If swapToken does not implement `permit` to the specifications of EIP-2612, this function will fail.
+     * @param swapToken Address of the token that will be swapped for acrossInputToken.
+     * @param acrossInputToken Address of the token that will be bridged via Across as the inputToken.
+     * @param routerCalldata ABI encoded function data to call on router. Should form a swap of swapToken for
+     * enough of acrossInputToken, otherwise this function will revert.
+     * @param swapTokenAmount Amount of swapToken to swap for a minimum amount of depositData.inputToken.
+     * @param minExpectedInputTokenAmount Minimum amount of received depositData.inputToken that we'll submit bridge
+     * deposit with.
+     * @param depositData Specifies the Across deposit params we'll send after the swap.
+     */
+    function swapAndBridgeWithPermit(
+        IERC20Permit swapToken,
+        IERC20 acrossInputToken,
+        bytes calldata routerCalldata,
+        uint256 swapTokenAmount,
+        uint256 minExpectedInputTokenAmount,
+        DepositData calldata depositData,
+        uint256 deadline,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) external nonReentrant {
+        swapToken.permit(msg.sender, address(this), swapTokenAmount, deadline, v, r, s);
+        _swapAndBridge(
+            routerCalldata,
+            swapTokenAmount,
+            minExpectedInputTokenAmount,
+            depositData,
+            IERC20(address(swapToken)), // Cast IERC20Permit to IERC20
+            acrossInputToken
+        );
+    }
+
+    /**
+     * @notice Swaps an EIP-3009 token on this chain via specified router before submitting Across deposit atomically.
+     * Caller can specify their slippage tolerance for the swap and Across deposit params.
+     // * @dev If swapToken does not implement `permit` to the specifications of EIP-2612, this function will fail. 
```

Comment:
Need to edit this comment

Reply:
https://github.com/across-protocol/contracts/pull/751/commits/7bdf27a7ab2d78508e8e86ef519b5fdf63b99ff9